smartdashboard/view_builders.py

```diff
@@ -466,3 +471,164 @@ def overview_builder(manifest: Manifest) -> OverviewView:
+
+
+def memory_view_builder(shards: t.List[Shard]) -> MemoryView:
+    """Memory section of Database Telemetry page to be rendered
+
+    :param shards: Shards of the selected Orchestrator
+    :type shards: t.List[Shard]
+    :return: View of the memory portion of the DB Telemetry page
+    :rtype: MemoryView
+    """
+    with st.expander(label="Memory"):
+        col1, col2 = st.columns([6, 6])
+        with col1:
+            view = MemoryView(shards[0] if shards else None)
+            shard = st.selectbox(
+                "Select a shard:",
+                shards,
+                format_func=lambda shard: shard.name,
+                key="memory_shard",
+            )
+            view.update_shard(shard)
+            view.memory_table_element = st.empty()
+        with col2:
+            st.write("")
+            st.write("")
+            st.write("")
+            view.memory_graph_element = st.empty()
+
+    return view
+
+
+def client_view_builder(shards: t.List[Shard]) -> ClientView:
+    """Client section of Database Telemetry page to be rendered
+
+    :param shards: Shards of the selected Orchestrator
+    :type shards: t.List[Shard]
+    :return: View of the client portion of the DB Telemetry page
+    :rtype: ClientView
+    """
+    with st.expander(label="Clients"):
+        col1, col2 = st.columns([6, 6])
+        with col1:
+            view = ClientView(shards[0] if shards else None)
+            shard = st.selectbox(
+                "Select a shard:",
+                shards,
+                format_func=lambda shard: shard.name,
+                key="client_shard",
+            )
+            view.update_shard(shard)
+            view.client_table_element = st.empty()
+        with col2:
+            st.write("")
+            st.write("")
+            st.write("")
+            view.client_graph_element = st.empty()
+
+    return view
+
+
+def orc_summary_builder(
+    selected_orchestrator: t.Optional[Orchestrator],
+) -> OrchestratorSummaryView:
+    """Orchestrator summary section of Database Telemetry page to be rendered
+
+    :param selected_orchestrator: Selected Orchestrator
+    :type selected_orchestrator: t.Optional[Orchestrator]
+    :return: View of the summary portion of the DB Telemetry page
+    :rtype: OrchestratorSummaryView
+    """
+    with st.expander(label="Orchestrator Summary"):
+
+        view = OrchestratorSummaryView(selected_orchestrator)
+
+        st.write("")
+        view.status_element = st.empty()
+        st.write(
+            "Number of shards: "
+            + (str(len(selected_orchestrator.shards)) if selected_orchestrator else "")
+        )
+        st.write(
+            "Type: "
+            + (selected_orchestrator.type if selected_orchestrator is not None else "")
+        )
+        st.write("Port: " + get_port(selected_orchestrator))
+        st.write(
+            "Interface: "
+            + (
+                ", ".join(selected_orchestrator.interface)
```

Comment:
What do you think about shifting any "computed fields" into the views to reduce the amount of places we have to look for formatting? I think we could do the same thing here as `get_port`

Reply:
created `format_interfaces` that's used exactly like `get_port` so we aren't computing anything in `view_builders`